pandas/_libs/join.pyx

```diff
@@ -138,7 +143,13 @@ def left_outer_join(const int64_t[:] left, const int64_t[:] right,
             # this is a short-cut to avoid groupsort_indexer
             # otherwise, the `else` path also works in this case
             rev = np.empty(len(left), dtype=np.intp)
-            rev.put(ensure_platform_int(left_sorter), np.arange(len(left)))
+            rev.put(
+                ensure_platform_int(left_sorter),
+                # NOTE:
+                # this is the C-optimized equivalent of
+                # `np.arange(len(left))`
+                cnp.PyArray_Arange(0, len(left), 1, NPY_INTP),
+            )
```

Comment:
@MomIsBestFriend this looks a lot like just `rev = ensure_platform_int(left_sorter)`.  am i missing something?

Reply:
You are absolutely right!

by the look of it at least.